gnodeb/gnodeb.go

```diff
@@ -60,20 +62,34 @@ func Init(gnb *gnbctx.GNodeB) error {
 
 	gnb.DefaultAmf.Init()
 
-	err = gnb.CpTransport.ConnectToPeer(gnb.DefaultAmf)
-	if err != nil {
-		gnb.Log.Errorln("ConnectToPeer returned:", err)
-		return fmt.Errorf("failed to connect to amf")
-	}
-
-	successfulOutcome, err := PerformNgSetup(gnb, gnb.DefaultAmf)
-	if !successfulOutcome || err != nil {
-		gnb.Log.Errorln("PerformNgSetup returned:", err)
-		return fmt.Errorf("failed to perform ng setup procedure")
-	}
-
-	go gnb.CpTransport.ReceiveFromPeer(gnb.DefaultAmf)
+	amfConnStatus := make(chan bool, 2)
+	amfConnStatus <- false
+
+	go func(amfConnStatus chan bool) {
+		for {
+			select {
+			case status := <-amfConnStatus:
+				gnb.Log.Infoln("Conn status received")
+				if !status {
+					err = gnb.CpTransport.ConnectToPeer(gnb.DefaultAmf)
+					if err != nil {
+						gnb.Log.Errorln("ConnectToPeer returned:", err)
+						return
+					}
+
+					successfulOutcome, err := PerformNgSetup(gnb, gnb.DefaultAmf)
+					if !successfulOutcome || err != nil {
+						gnb.Log.Errorln("PerformNgSetup returned:", err)
+						return
+					}
+
+					go gnb.CpTransport.ReceiveFromPeer(gnb.DefaultAmf, amfConnStatus)
+				}
+			}
+		}
+	}(amfConnStatus)
 
+	time.Sleep(10 * time.Second)
```

Comment:
Expectation of the InitializeAllGnbs function is to connect to gNB and return back. What do we achieve by adding 10 seconds sleep here ? Also once code returned then this go routine continues to run ?

Reply:
sometimes, I was  seeing that this connect to peer was not ready and the test case started. So added some delay here for the connection to be established.